autoPyTorch/datasets/base_dataset.py

```diff
@@ -106,14 +108,16 @@ def __init__(
             val_transforms (Optional[torchvision.transforms.Compose]):
                 Additional Transforms to be applied to the validation/test data
         """
-        self.dataset_name = dataset_name if dataset_name is not None \
-            else hash_array_or_matrix(train_tensors[0])
+        self.dataset_name = ""
+
+        if self.dataset_name is not None:
```

Comment:
I think self.dataset_name is always going to be None due to line 111?

Can we make dataset_name not Optional to simplify the code?